internal/controller/helmrelease_controller_chart.go

```diff
@@ -118,7 +118,12 @@ func (r *HelmReleaseReconciler) loadHelmChart(source *sourcev1b2.HelmChart) (*ch
 	defer f.Close()
 	defer os.Remove(f.Name())
 
-	artifactURL := source.GetArtifact().URL
+	artifact := source.GetArtifact()
+	if artifact == nil {
+		return nil, fmt.Errorf("source '%s/%s' has no artifact", source.GetNamespace(), source.GetName())
```

Comment:
```suggestion
		return nil, fmt.Errorf("can not load Helm chart: source '%s/%s' has no artifact", source.GetNamespace(), source.GetName())
```

would probably be more descriptive.